packages/vite/src/node/server/index.ts

```diff
@@ -896,7 +896,9 @@ async function startServer(
   server._currentServerPort = serverPort
 }
 
-function createServerCloseFn(server: HttpServer | null) {
+export function createServerCloseFn(
+  server: HttpServer | null,
+): () => Promise<void> | void {
```

Comment:
maybe we could make this function return `Promise<void>` only and in 903 return an async function so you don't need to do the `as` here https://github.com/vitejs/vite/pull/15630/files#diff-35ba301b85014a4bfaa9cad2d8e7eafa41c4e8c2ddd5c193182241d9a1542082R135

Reply:
Change line 903 directly to `() => Promise.resolve()` ?